src/cpp/astar.cpp

```diff
@@ -36,6 +38,44 @@ inline float l1_norm(int i0, int j0, int i1, int j1) {
   return std::abs(i0 - i1) + std::abs(j0 - j1);
 }
 
+// Please note below heuristic is experimental and only for pretty lines
```

Comment:
Thanks for adding this warning. Let's move these two experimental heuristics to a separate file and include this warning at the top.

Reply:
sure, makes sense. my c++ is lacking will check how to do that. probably 10years ago last i actually looked at c++.

Reply:
Thanks. Let me know if you get stuck on this. Because we inline the heuristic functions I think you'll have to implement them in the header file.